util/dvsim/Testplan.py

```diff
@@ -0,0 +1,686 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+r"""Testpoint and Testplan classes for maintaining the testplan
+"""
+
+import os
+import re
+import sys
+
+import hjson
+import mistletoe
+from tabulate import tabulate
+
+
+class TestResult:
+    '''The results for a single test'''
+    def __init__(self, name, passing=0, total=0):
+        self.name = name
+        self.passing = passing
+        self.total = total
+        self.mapped = False
+
+
+class TestplanElement():
+    """An element of the testplan.
+
+    This is either a testpoint or a covergroup.
+    """
+    # Type of the testplan element. Must be set by the extended class.
+    kind = None
+
+    # Mandatory fields in a testplan element.
+    fields = ("name", "desc")
+
+    def __init__(self, raw_dict):
+        """Initialize the testplan element.
+
+        raw_dict is the raw dict parsed from the HJSon file.
+        """
+        try:
+            for field in self.fields:
+                setattr(self, field, raw_dict.pop(field))
+        except KeyError:
+            raise KeyError(f"Error: {self.kind.capitalize()} does not contain "
```

Comment:
Could you mention the missing one that triggered the exception? And perhaps put the try within the loop?